File: BackEnd - ControleDeVacinacaoBovina/Models/Animal.cs
using System;

namespace ControleDeVacinacaoBovina.Models
{
    public class Animal
    {
        public int IdAnimal { get; set; }
        public int QuantidadeTotal { get; set; }
        public int QuantidadeVacinada { get; set; }
        public int IdEspecie { get; set; }
        public int IdPropriedade { get; set; }
        public int IdTipoDeEntrada { get; set; }
        public DateTime DataDeEntrada { get; set; }
        public bool Ativo { get; private set; } = true;
        public TipoDeEntrada TipoDeEntrada { get; set; }
        public Especie Especie { get; set; }
        public Propriedade Propriedade { get; set; }

        public void SetAtivo(bool ativo)
        {
            Ativo = ativo;
        }
    }
}

File: FrontEnd - ControleDeVacinacaoBovina/angular-controle-de-vacinacao-bovina/src/app/entities/animal.ts
import { Especie } from './especie';
import { Propriedade } from "./propriedade";
import {tipoDeEntradas} from "./tipodeentradas";

export class Animal {
	idAnimal: number = 0;
	quantidadeTotal?: number;
	quantidadeVacinada?: number;
	idEspecie: number = 0;
	idPropriedade: number = 0;
  idTipoDeEntrada: number = 0;
	ativo: boolean = true;
	propriedade: Propriedade = new Propriedade();
  especie: Especie = new Especie();
  tipoDeEntrada: tipoDeEntradas = new tipoDeEntradas();
}
